packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -755,19 +766,36 @@ function updateReducer<S, I, A>(
     let newState = baseState;
     let newBaseState = null;
     let newBaseUpdate = null;
+    let newRebaseTime = NoWork;
+    let newRebaseEnd = null;
     let prevUpdate = baseUpdate;
     let update = first;
-    let didSkip = false;
+
+    // Track whether the update is part of a rebase.
+    // TODO: Should probably split this into two separate loops, instead of
+    // using a boolean.
+    let isRebasing = rebaseTime !== NoWork;
+
     do {
       const updateExpirationTime = update.expirationTime;
-      if (updateExpirationTime < renderExpirationTime) {
+      if (prevUpdate === rebaseEnd) {
+        isRebasing = false;
+      }
+      if (
+        // Check if this update should be skipped
+        updateExpirationTime < renderExpirationTime &&
+        // If we're currently rebasing, don't skip this update if we already
+        // committed it.
+        (!rebasedUpdatesNeverUncommit ||
+          (!isRebasing || updateExpirationTime < rebaseTime))
+      ) {
         // Priority is insufficient. Skip this update. If this is the first
         // skipped update, the previous update/state is the new base
         // update/state.
-        if (!didSkip) {
-          didSkip = true;
+        if (newRebaseTime === NoWork) {
           newBaseUpdate = prevUpdate;
           newBaseState = newState;
+          newRebaseTime = renderExpirationTime;
```

Comment:
Ooh I think this should be the lower of `renderExpirationTime` and the old `rebaseTime`. In case there are two rebases in a row. I'll add a new test.

Reply:
It would also have to be the further at of the RebaseEnd then, no?

Reply:
Actually I don't think this will work. There can be arbitrary numbers of rebases, each with their own ranges and expiration times.

Will need to evaluate.